ptvsd/wrapper.py

```diff
@@ -853,7 +848,9 @@ def _handle_disconnect(self, request):
         self.disconnect_request_event.set()
         self._notify_disconnecting(not self._closed)
         if not self._closed:
-            self.close()
+            # Closing the socket causes pydevd to resume all threads,
+            # so just terminate the process altogether.
+            sys.exit(0)
```

Comment:
Won't this prevent re-attach?

Reply:
It will, but in this scenario (launch), there's not supposed to be one - disconnect after launch means terminate debuggee, or at least that's how we have always implemented it so far in the old debugger.

Reply:
Ah, right, ``on_disconnect()`` doesn't call ``_handle_disconnect()`` in the "attach" case.